### PR TITLE
RavenDB-20377 Add mixed state to index status

### PR DESCRIPTION
**Pull request by mateuszbartosik**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-20377/Add-mixed-state-to-the-Index-status

### Additional description
Added some disabled, some errored, some paused, some faulty, some pending statuses. Moved the expand/collapse icon to the left.

### Type of change
- Optimization

### How risky is the change?
- Low 

### Backward compatibility
- Not relevant

### Is it platform specific issue?
- No

### Documentation update
- No documentation update is needed 

### Testing by Contributor
- It has been verified by manual testing

### Testing by RavenDB QA team
- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx [hunk 1]
```diff
@@ -190,68 +191,96 @@ function calculateOverallProgress(index: IndexSharedInfo) {
...
-        const overallProgress = calculateOverallProgress(index);
-
-        return (
-            <ProgressCircle inline state="running" progress={overallProgress}>
-                Running
-            </ProgressCircle>
-        );
-    }
+    const { index, onClick } = props;
+    const getState = (index: IndexSharedInfo) => {
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx]: if this is pure function move it outside

File: src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx [hunk 2]
```diff
@@ -190,68 +191,96 @@ function calculateOverallProgress(index: IndexSharedInfo) {
...
+                    x.details?.status === "Pending" ||
+                    x.details?.stale
+            )
+        ) {
+            return "running";
+        }
+        return "success";
+    };
+
+    const getIcon = (index: IndexSharedInfo): IconName => {
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx]: same here

File: src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx [hunk 3]
```diff
@@ -190,68 +191,96 @@ function calculateOverallProgress(index: IndexSharedInfo) {
...
+            return iconForState("Pending");
+        }
+        if (index.nodesInfo.some((x) => x.details?.stale)) {
+            return null;
+        }
+
+        return "check";
+    };
+
+    const getStateText = (index: IndexSharedInfo) => {
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx]: same

File: src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx [hunk 4]
```diff
@@ -190,68 +191,96 @@ function calculateOverallProgress(index: IndexSharedInfo) {
...
+            return "Load errors";
+        }
+        if (index.nodesInfo.every((x) => x.details?.faulty)) {
+            return "Faulty";
+        }
+        if (index.nodesInfo.some((x) => x.details?.faulty)) {
+            return "Some faulty";
+        }
+        if (index.nodesInfo.every((x) => x.details?.state === "Error")) {
+            return "Errors";
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx]: it's plural?

**Review by kalczur — CHANGES_REQUESTED** [review]

File: src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx (hunk 1 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx]: Done

File: src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx (hunk 4 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/pages/database/indexes/list/IndexDistribution.tsx]: It makes sense for multiple errors:
image

**Review by ml054 — APPROVED** [review]

**Pull request merged**